File: 백준/Silver/19637. IF문 좀 대신 써줘/IF문 좀 대신 써줘.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st1 = new StringTokenizer(br.readLine());
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(st1.nextToken());
        int M = Integer.parseInt(st1.nextToken());

        String[] level = new String[N];
        int[] power = new int[N];

        for (int i = 0; i < N; i++) {
            StringTokenizer st2 = new StringTokenizer(br.readLine());
            level[i] = st2.nextToken();
            power[i] = Integer.parseInt(st2.nextToken());
        }

        for (int i = 0; i < M; i++) {
            int num = Integer.parseInt(br.readLine());

            int left = 0;
            int right = N;
            int res = 0;

            while (left <= right) {
                int mid = (left + right) / 2;
                if (power[mid] >= num) {
                    res = mid;
                    right = mid - 1;
                } else {
                    left = mid + 1;
                }
            }

            sb.append(level[res]).append("\n");
        }

        System.out.println(sb.toString());
    }
}

File: 백준/Silver/19637. IF문 좀 대신 써줘/IF문 좀 대신 써줘.py
import sys

N, M = map(int, sys.stdin.readline().split(' '))
level = []
power = []

for i in range(N):
    A, B = map(str, sys.stdin.readline().split(' '))
    # # 전투력은 내림차순으로 주어지고, 중복되면 먼저 입력된 칭호만 출력해야 하므로 중복 방지
    # if (int(B) not in power):
    #     level.append(A)
    #     power.append(int(B))
    level.append(A)
    power.append(int(B))

arr = []
for i in range(M):
    arr.append(int(sys.stdin.readline().strip()))

for i in arr :
    left = 0
    right = len(power)
    res = 0

    while left <= right :
        mid = (left + right) // 2
        if power[mid] >= i :
            res = mid
            right = mid - 1
        else :
            left = mid + 1
    print(level[res])
